src/backend/access/transam/xlog.c

```diff
@@ -8207,7 +8211,8 @@ StartupXLOG(void)
 	if (fast_promoted)
 		RequestCheckpoint(CHECKPOINT_FORCE);
 
-	*shmCleanupBackends = true;
+	if (Gp_role == GP_ROLE_DISPATCH)
+		*shmCleanupBackends = true;
```

Comment:
Maybe it's worth replacing the check with an Assert that the pointer is not NULL?

Reply:
Assert works only in debug build.

Reply:
That's the point. If we now assume that this cannot happen, what is the point of checking this in the release build?